hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeAttributeProvider.java

```diff
@@ -68,6 +277,8 @@ public abstract void checkPermission(String fsOwner, String supergroup,
         boolean ignoreEmptyDir)
             throws AccessControlException;
 
+    void checkPermissionWithContext(AuthorizationContext authzContext)
```

Comment:
NIT: can you add javadoc for the new public method?

Reply:
done.